pinot-spi/src/main/java/org/apache/pinot/spi/data/SchemaValidator.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.data;
+
+
+/**
+ * Validator to validate the schema between Pinot schema and input raw data schema
+ */
+public interface SchemaValidator {
+
+  void init(Schema pinotSchema, String inputFilePath);
+
+  boolean isDataTypeMismatch();
```

Comment:
If we return the field names that have this type of mismatch, and also an error message (e.g. "input type 'List' does not match with pinot data type 'int' for field 'X') that will be awesome.
We can then log this as an error message during segment creation

Reply:
Added a class called `SchemaValidatorResult` to provide detailed information. I've added the sample detailed info in the description of this PR.